client/src/App.js

```diff
@@ -0,0 +1,38 @@
+import './App.css';
+import React, {useState} from 'react';
+import UserProfile from './component/UserProfile';
+import {BrowserRouter as Router, Route, Redirect, Switch, NavLink} from 'react-router-dom';
+import Login from './component/Login';
+import Home from './component/Home';
+import { useHistory } from 'react-router';
+import loginContext from './context/loginContext';
+
+function App() {
+  const [userName, setUserName] = useState("");
+  const [password, setPassword] = useState("");
+  const [user, setUser] = useState({});
+  const [invalidUser, setInvalidUser] = useState("");
+  const history = useHistory();
+
+  
+  return (
+    <div className="App">
+  
+  <loginContext.Provider value={{userName, setUserName, password, setPassword, user, setUser, invalidUser, setInvalidUser, history}} >
```

Comment:
The code indentation in this file is a bit chaotic, typically the project manager would complain about this :) 
But sometimes I forgot to do the code formatting too before pushing the code... so fine this time ;)